### DISCUSSION TITLE
Old NEXRAD "missing" lat lon

### DISCUSSION THREAD

**Discussion by dopplerchase** [ARM-DOE/pyart#1174] [Q&A] [answered]
I was playing around with some old NEXRAD data (Sept 27 1998 from KBUF; 4 inch hail day) and I noticed that these old files have their latitude and longitude a bit messed up. It seems to be centered on the radar location 

![image](https://user-images.githubusercontent.com/22037322/173164544-6735e23d-04eb-4bd1-b61f-7b93d9ea3fa7.png)

Reproduce the issue here: 

https://colab.research.google.com/drive/1PeFXuAez7P7pXbkYixTys0pawevSuyKm?usp=sharing

I think its a decently easy fix if you can parse the name of the file and then get the lat,lon of the radar location. 

Thats what i did as a work around:
```
    rad_lon=-78.73667
    rad_lat=42.94889
    lon = rad_lon + radar.gate_longitude['data'][s_idx:e_idx,:]
    lat = rad_lat +radar.gate_latitude['data'][s_idx:e_idx,:]
```

has this already been implemented somewhere in pyart? if not, where do you see it being implemented? (so I can maybe start drafting a function to help fix)

**Comment by mgrover1** [comment]
Great question @dopplerchase - we currently do not make any assumptions about reading the lat/lon from the filename, but you can pass the station id into the [`pyart.io.read_nexrad_archive`](https://github.com/ARM-DOE/pyart/blob/main/pyart/io/nexrad_archive.py#L19) function, as `station="KBUF"`. We have functions that are able to index the name of the radar, and use the lats and lons of the radar site. 

As is the case with the older data, if it can't find the radar lat and lon, it will set it to 0, which is what happened here.

**Comment by mgrover1** [comment, marked as answer]
The IO line would look something like this:

```python
import pyart 
radar = pyart.io.read('./KBUF19980927_204203.gz', station="KBUF")
```
